src/display.rs

```diff
@@ -154,24 +154,82 @@ pub struct DisplayByteSlice<'a> {
 
 impl<'a> DisplayByteSlice<'a> {
     fn display(&self, f: &mut fmt::Formatter, case: Case) -> fmt::Result {
+        use fmt::Write;
+        // There are at least two optimizations left:
+        //
+        // * Reusing the buffer (encoder) which may decrease the number of virtual calls
+        // * Not recursing, avoiding another 1024B allocation and zeroing
+        //
+        // This would complicate the code so I was too lazy to do them but feel free to send a PR!
+
         let mut buf = [0u8; 1024];
         let mut encoder = BufEncoder::new(&mut buf);
 
-        // Its unlikely that someone will want special formatting for a hex string that
-        // is over 1024 characters so just handle padding for short slices.
-        if self.bytes.len() < 512 {
-            encoder.put_bytes(self.bytes, case);
-            return f.pad(encoder.as_str());
+        let pad_right = if let Some(width) = f.width() {
+            let string_len = match f.precision() {
+                Some(max) if self.bytes.len() * 2 > (max + 1) / 2 => max,
+                Some(_) | None => self.bytes.len() * 2,
+            };
+
+            if string_len < width {
+                let (left, right) = match f.align().unwrap_or(fmt::Alignment::Left) {
+                    fmt::Alignment::Left => (0, width - string_len),
+                    fmt::Alignment::Right => (width - string_len, 0),
+                    fmt::Alignment::Center =>
+                        ((width - string_len) / 2, (width - string_len + 1) / 2),
+                };
+                // Avoid division by zero and optimize for common case.
```

Comment:
In 8569ee8e55da4ecb099c68dca7eceb9ebae619ba:

Here and below I don't think there is any actual division by 0. There would be "dividing 0 by something" but this is harmless.

Reply:
`chunk_len` is 0 if the input to `put_filler` is 0. I actually got panics for that, unfortunately I wasn't smart enough to see it first. :)

Reply:
Lol, me neither apparently :)